tmp/lanelet2_extension/lib/utilities.cpp

```diff
@@ -501,12 +501,28 @@ lanelet::ConstLanelets getExpandedLanelets(
 }
 
 void overwriteLaneletsCenterline(
-  lanelet::LaneletMapPtr lanelet_map, const double resolution, const bool force_overwrite)
+  lanelet::LaneletMapPtr lanelet_map, const double resolution, const bool use_waypoints,
+  const bool force_overwrite)
 {
   for (auto & lanelet_obj : lanelet_map->laneletLayer) {
-    if (force_overwrite || !lanelet_obj.hasCustomCenterline()) {
+    if (force_overwrite) {
       const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
       lanelet_obj.setCenterline(fine_center_line);
+    } else {
+      if (use_waypoints) {
+        if (lanelet_obj.hasCustomCenterline()) {
+          const auto & centerline = lanelet_obj.centerline();
+          lanelet_obj.setAttribute("waypoints", centerline.id());
+        }
+
+        const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
+        lanelet_obj.setCenterline(fine_center_line);
+      } else {
+        if (!lanelet_obj.hasCustomCenterline()) {
+          const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
+          lanelet_obj.setCenterline(fine_center_line);
+        }
+      }
     }
```

Comment:
Maybe you can simplify the nesting
```suggestion
    if (force_overwrite) {
      const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
      lanelet_obj.setCenterline(fine_center_line);
    } else if (use_waypoints) {
      if (lanelet_obj.hasCustomCenterline()) {
        const auto & centerline = lanelet_obj.centerline();
        lanelet_obj.setAttribute("waypoints", centerline.id());
      }

      const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
      lanelet_obj.setCenterline(fine_center_line);
    } else if (!lanelet_obj.hasCustomCenterline()) {
      const auto fine_center_line = generateFineCenterline(lanelet_obj, resolution);
      lanelet_obj.setCenterline(fine_center_line);
    }
```

Reply:
The original code looks better to me, but my opinion is not strong at all.
I will apply your change. Thank you.